include/libtorrent/session.hpp

```diff
@@ -147,23 +158,37 @@ namespace libtorrent
 	{
 	public:
 
+		session(session_params params = session_params())
+			: session_handle(nullptr)
+		{
+			TORRENT_CFG();
+			start(std::move(params), nullptr);
+		}
+
+		session(session_params params, io_service& ios)
+			: session_handle(nullptr)
+		{
+			TORRENT_CFG();
+			start(std::move(params), &ios);
+		}
+
 		// Constructs the session objects which acts as the container of torrents.
 		// It provides configuration options across torrents (such as rate limits,
 		// disk cache, ip filter etc.). In order to avoid a race condition between
 		// starting the session and configuring it, you can pass in a
 		// settings_pack object. Its settings will take effect before the session
 		// starts up.
-		// 
+		//
 		// The ``flags`` parameter can be used to start default features (upnp &
 		// nat-pmp) and default plugins (ut_metadata, ut_pex and smart_ban). The
 		// default is to start those features. If you do not want them to start,
 		// pass 0 as the flags parameter.
-		session(settings_pack pack = settings_pack()
-			, int flags = start_default_features | add_default_plugins)
+		session(settings_pack pack
+			, bool default_plugins = true)
```

Comment:
oh, I didn't notice you changed the flags to a bool. In general I would prefer flags over a bool, because it makes the call site much more readable.

Regarding the constructor, I'm a bit hesitant to remove or deprecate the existing constructor. I think we should keep it for backwards compatibility and add a new one that's well thought through, and that can survive longer. Maybe instead of specifying a default plugin flag, the vector of extensions can be pre-populated with the default ones, for instance.